src/Microsoft.Cci.Extensions/Extensions/CSharp/CSharpCciExtensions.cs

```diff
@@ -333,6 +333,30 @@ public static bool IsDestructor(this IMethodDefinition methodDefinition)
             return false;
         }
 
+        public static bool IsDispose(this IMethodDefinition methodDefinition)
+        {
+            if (methodDefinition.Name.Value != "Dispose")
+            {
+                return false;
+            }
+
+            if (methodDefinition.ParameterCount == 0 && methodDefinition.Visibility == TypeMemberVisibility.Public &&
+                TypeHelper.TypesAreEquivalent(methodDefinition.Type, methodDefinition.ContainingTypeDefinition.PlatformType.SystemVoid))
+            {
+                // Dispose() should be public void.
```

Comment:
This isn't necessarily true. I'm guessing we have some explicit implementations of IDisposable which means the method will not be public nor will it be named Dispose it will be named "IDisposable.Dispose" and private.

Reply:
I think I would lossen the check to make sure the name is Dispose or IDisposable.Dispose and returns void. That will allow other things it but for our usage should be fine.

Reply:
So you don't think it is worth checking for parameterCount and if ParameterCount == 1 check that the parameter type is boolean either?

Reply:
probably not worth checking the parameters or visibility of the method.

Reply:
I'm not opposed to checking for the bool parameter if you want to keep it but you should fix name check to support explicit interface implementations and that will require you to remove the visibility check.

Reply:
Ok.

Reply:
thinking about this I think the explicit implementation name is "System.IDisposable.Dispose" but please verify.

Reply:
Yes, you are right.

Reply:
Fixed.